File: android/app/src/main/java/com/teameetmeet/meetmeet/data/network/entity/EventResponse.kt
package com.teameetmeet.meetmeet.data.network.entity

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.teameetmeet.meetmeet.data.model.EventMember

@JsonClass(generateAdapter = true)
data class EventResponse(
    @Json(name = "id")
    val id: Int,
    @Json(name = "title")
    val title: String,
    @Json(name = "startDate")
    val startDate: String,
    @Json(name = "endDate")
    val endDate: String,
    @Json(name = "eventMembers")
    val eventMembers: List<EventMember> = listOf(),
    @Json(name = "authority")
    val authority: String = "",
    @Json(name = "repeatPolicyId")
    val repeatPolicyId: Int? = null,
    @Json(name = "isJoinable")
    val isJoinable: Boolean = false,
    @Json(name = "alarmMinutes")
    val alarmMinutes: Int = -1,
    @Json(name = "color")
    val color: Int = -39579
)

@JsonClass(generateAdapter = true)
data class Events(
    @Json(name = "events")
    val events: List<EventResponse>
)

@JsonClass(generateAdapter = true)
data class SingleEvent(
    @Json(name = "event")
    val event: EventResponse
)

File: backend/src/event/dto/event-response.ts
import { ApiProperty } from '@nestjs/swagger';
import { EventMember } from 'src/event-member/entities/eventMember.entity';
import { UserResponse } from 'src/user/dto/user-response';

export class Member extends UserResponse {
  @ApiProperty()
  authority: string;

  static of(eventMember: EventMember): Member {
    return {
      id: eventMember.user.id,
      nickname: eventMember.user.nickname,
      profile: eventMember.user.profile?.path ?? null,
      authority: eventMember.authority.displayName,
    };
  }
}

export class EventResponse {
  @ApiProperty()
  id: number;
  @ApiProperty()
  title: string;
  @ApiProperty()
  startDate: string;
  @ApiProperty()
  endDate: string;
  @ApiProperty({ type: Member })
  eventMembers: Member[];
  @ApiProperty()
  authority: string | null;
  @ApiProperty()
  isJoinable: boolean;
}
